External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -2319,11 +2332,35 @@ static public void HandleAllClassesCompletion(ScintillaControl Sci, string tail,
                 }
             }
 
-            if (ASContext.Context.Features.hasDelegates && !ASContext.Context.CurrentClass.IsVoid())
+            if (!ASContext.Context.CurrentClass.IsVoid())
             {
-                foreach (MemberModel field in ASContext.Context.CurrentClass.Members)
-                    if ((field.Flags & FlagType.Delegate) > 0)
-                        known.Add(field);
+                if (ASContext.Context.Features.hasDelegates)
+                {
+                    MemberList delegates = new MemberList();
+
+                    foreach (MemberModel field in ASContext.Context.CurrentClass.Members)
+                        if ((field.Flags & FlagType.Delegate) > 0)
+                            delegates.Add(field);
+
+                    if (delegates.Count > 0)
+                    {
+                        delegates.Sort();
+                        delegates.Merge(known);
+                        known = delegates;
+                    }
+                }
+
+                if (ASContext.Context.Features.hasGenerics)
+                {
+                    var typeParams = GetVisibleTypeParameters();
+
+                    if (typeParams != null && typeParams.Items.Count > 0)
+                    {
+                        typeParams.Add(known);
```

Comment:
Merging 2 sorted lists should be the fastest - it does a single iteration of the lists and inserts the items where they belong.